### ISSUE TITLE
Task `:buildDll` fails on Linux

### ISSUE THREAD

**Issue by Rayzr522** [NovaMods/nova-renderer#68]
Going off of the updated instructions that you told me of in #38, I'm was attempting to install the `chunks` branch via `./gradlew setup` and `./gradlew runNova`. When I run `./gradlew runNova`, this is what I get:

```
$ ./gradlew runNova
:compileJava UP-TO-DATE
:processResources UP-TO-DATE
:classes UP-TO-DATE
:jar UP-TO-DATE
:assemble UP-TO-DATE
:compileTestJava UP-TO-DATE
:processTestResources UP-TO-DATE
:testClasses UP-TO-DATE
:test UP-TO-DATE
:check UP-TO-DATE
:build UP-TO-DATE
:buildDll FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':buildDll'.
> A problem occurred starting process 'command 'powershell''

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 1.995 secs
```

A quick inspection of your build files reveals this:

```
$ spot powershell
  
./build.gradle:  
32:    commandLine 'powershell', '-file', 'nova.ps1', '-build', '-nativeOnly' 
```

So your `build.gradle` file is running `nova.ps1`, even though you stated in #38 that it's now deprecated.

**Comment by DethRaid** [comment]
Guess I forgot about that. Maybe nova.sh isn't deprecated quite yet

**Comment by DethRaid** [comment]
We're just committed some changes to the build script. Could you test again and see if Nova works for you?

**Comment by Rayzr522** [comment]
Alright, so other than a slight tweak to the `setup.gradle` file (#104), the setup process now works completely. And after quite a bit of effort, I got the VulkanSDK set up on Linux. However, from a clean installation, I get this problem with the `buildDll` stage: 

```
$ ./gradlew buildDll
Configuring for GCC
:invokeCMake
-- Using X11 for window creation
-- GLM is a header only library, no need to build it. Set the option GLM_TEST_ENABLE with ON to build and run the test bench
GLM: GCC - GNU compiler
-- No build type selected, default to Debug
-- Google Mock already configured
-- SPIRV-Tools: Effcee is not configured.  Skipping Effcee-based tests.
-- Found Google Mock, building tests.
-- Google Mock already configured - use it
-- optimizer enabled
-- Google Mock found - building tests
-- Shaderc: build type is "".
-- Configuring Shaderc to build tests.
-- asciidoctor was not found - no documentation will be generated
-- Configuring done
-- Generating done
-- Build files have been written to: /home/rayzr/Desktop/nova-renderer/src/main/cpp
:buildDll
[  3%] Built target HLSL
[ 30%] Built target SPIRV-Tools
[ 53%] Built target SPIRV-Tools-opt
[ 53%] Built target OSDependent
[ 53%] Built target OGLCompiler
[ 66%] Built target glslang
[ 66%] Built target SPVRemapper
[ 70%] Built target SPIRV
[ 76%] Built target shaderc_util
[ 76%] Built target shaderc
[ 93%] Built target nova-renderer-obj
[100%] Built target glfw
make[3]: *** No rule to make target '3rdparty/spirv-tools/source/opt/libSPIRV-Tools-opt.a', needed by 'libnova-renderer.so'.  Stop.
make[2]: *** [CMakeFiles/Makefile2:79: CMakeFiles/nova-renderer.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:91: CMakeFiles/nova-renderer.dir/rule] Error 2
make: *** [Makefile:164: nova-renderer] Error 2
:buildDll FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':buildDll'.
> Process 'command 'make'' finished with non-zero exit value 2

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 1.361 secs
```

I checked the Makefile in `src/main/cpp` and confirmed, there is no target with that name. 

This looks like the last little bug to fix before the system is completely functional. I look forward to being able to try out Nova :)

**Comment by DethRaid** [comment]
It seems like the build script outputs a debug version of that library, when a release  version is needed. You can manually rename `3rdparty/spirv-tools/source/opt/libSPIRV-Tools-optd.a` to `3rdparty/spirv-tools/source/opt/libSPIRV-Tools-opt.a` for now. That library is a third party dependency (as the folder it's in implies) so it will hardly ever change, so you won't have to do that very much

**Comment by Rayzr522** [comment]
Thanks, all fixed with #106.
